src/java.desktop/macosx/classes/com/apple/laf/AquaPainter.java

```diff
@@ -286,25 +295,4 @@ protected Image createTemplateImage(int width, int height) {
             return image;
         }
     }
-
-    private Graphics2D getGraphics2D(final Graphics g) {
-        try {
-            return (SunGraphics2D)g; // doing a blind try is faster than checking instanceof
```

Comment:
As far as I understand this was implemented this way due to performance reason, where the most common case for Sun Graphics2D was checked first, instead of 4 instanceof in the new code.

Reply:
That's also my impression, and I get what you mean by the 4 instanceof checks requiring more checks to be taken on average (assuming SG2D is the likeliest scenario, which it probably is). An additional check for SG2D placed at the top would mean fewer branches, but repeated code otherwise covered in the final block.

```java
if (g instanceof SunGraphics2D sg2d) {
    // SunGraphics is the likeliest graphics type provided (shortcut)
    paint(sg2d, nextState);
} else if (g instanceof PeekGraphics) {
    // if it is a peek just dirty the region
    g.fillRect(x, y, w, h);
} else if (g instanceof ProxyGraphics2D proxy && proxy.getDelegate() instanceof SunGraphics2D sg2d) {
    // if it is a proxy, check if the proxy is a SunGraphics2D
    paint(sg2d, nextState);
} else if (g instanceof Graphics2D g2d) {
    // otherwise any Graphics2D can be used to paint
    paint(g2d, nextState);
}
```

Reply:
I'm fairly certain @mrserb means that a raw cast and exception is faster than even an instanceof with the most common branch placed first in the code, I doubt this change is going to receive much support unfortunately

Reply:
Hmm... that's interesting. I'll have to do some testing on that.